### PR TITLE
feat: Adding authentication with options None and Custom

### LINKED ISSUE #47: Authentication: Custom
Add symbol for authentication with value custom

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): closes #47 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

No support for authentication in template

## What is the new behavior?

Adds authentication symbol with values none and custom

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by nickrandolph** [comment]
This is blocked waiting on this PR to merge: https://github.com/unoplatform/uno.extensions/pull/1362

File: src/Uno.Templates/content/unoapp/.template.config/template.json [hunk 1]
```diff
@@ -408,6 +408,39 @@
...
+        // {
+        //   "choice": "oidc",
+        //   "description": "Application uses OpenID Connect to authenticate user",
+        //   "displayName": "OIDC"
+        // },
+        // {
+        //   "choice": "Web",
+        //   "description": "Application uses web based authentication to authenticate user",
+        //   "displayName": "Web"
+        // }
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/.template.config/template.json:441]: may want to make this easier for your future PRs
```suggestion
        // },
        // {
        //   "choice": "msal",
        //   "description": "Application uses MSAL to authenticate user",
        //   "displayName": "MSAL"
        // },
        // {
        //   "choice": "oidc",
        //   "description": "Application uses OpenID Connect to authenticate user",
        //   "displayName": "OIDC"
        // },
        // {
        //   "choice": "Web",
        //   "description": "Application uses web based authentication to authenticate user",
        //   "displayName": "Web"
        }
```

File: src/Uno.Templates/content/unoapp/.template.config/template.json [hunk 2]
```diff
@@ -1340,6 +1386,57 @@
...
+      }
+    },
+    "authenticationEvaluator": {
+      "type": "generated",
+      "generator": "coalesce",
+      "parameters": {
+        "sourceVariableName": "authentication",
+        "fallbackVariableName": "presetAuthenticationDefault"
+      }
+    },
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/.template.config/template.json:1414]: I would tend to think this is easier to remove and just provide a default value for authentication... unless we need a different default value for recommended vs blank at some point

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/LoginPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#endif
+					new StackPanel()
+						.Grid(row: 1)
+						.HorizontalAlignment(HorizontalAlignment.Center)
+						.VerticalAlignment(VerticalAlignment.Center)
+						.Width(200)
+						.Spacing(16)
+						.Children(
+							new TextBox()
+								.Text(x => x.Bind(() => vm.Name).Mode(BindingMode.TwoWay))
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/LoginPage.xaml.cs:33]:
```suggestion
								.Text(x => x.Bind(() => vm.Name).TwoWay())
```

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/LoginPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+								.PlaceholderText("Username")
+								.HorizontalAlignment(HorizontalAlignment.Stretch),
+                            new PasswordBox()
+                                .Password(x => x.Bind(() => vm.Password).Mode(BindingMode.TwoWay))
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/LoginPage.xaml.cs:37]:
```suggestion
                                .Password(x => x.Bind(() => vm.Password).TwoWay())
```

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/MainPage.xaml.cs [hunk 5]
```diff
@@ -26,15 +26,22 @@ public MainPage()
...
-								.PlaceholderText("Enter your name:")
-								.Margin(8),
+								.PlaceholderText("Enter your name:"),
 							new Button()
 								.Content("Go to Second Page")
 								.AutomationProperties(automationId: "SecondPageButton")
-								.Command(() => vm.GoToSecond)))));
+								.Command(() => vm.GoToSecond)
+#if useAuthentication
+							new Button()
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/MainPage.xaml.cs:39]:
```suggestion
#if !useAuthentication
								.Command(() => vm.GoToSecond)
#else
								.Command(() => vm.GoToSecond),
							new Button()
```

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/App.recommended.cs [hunk 6]
```diff
@@ -102,7 +146,24 @@ protected async override void OnLaunched(LaunchActivatedEventArgs args)
 		_window.Activate();
 //+:cnd:noEmit
 #else
+#if (!useAuthentication)
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/App.recommended.cs:149]: Can we simplify this with else if and else?

**Review by dansiegel — APPROVED** [review]

**Pull request merged**